### ISSUE TITLE
Anchors are generated with the deprecated attribute `name` instead of `id`

### ISSUE THREAD

**Issue by marrws** [terraform-docs/terraform-docs#629]
### Describe the bug

- **Current behavior:** When generating docs with anchors, the html element `<a>` uses the deprecated attribute `name`
- **Expected behavior:** It should use the attribute `id` as the html5 spec says

This is what MDN[^1] says about `name`:
> Deprecated
>
> Was required to define a possible target location in a page. In HTML 4.01, `id` and `name` could both be used on `<a>`, as long as they had identical values.

This is what MDN says about `id`[^2] and fragment identification[^3]
> The `id` global attribute defines an identifier (ID) which must be unique in the whole document. Its purpose is to identify the element when linking (using a fragment identifier), scripting, or styling (with CSS).

### How can we reproduce it?

Generate docs with the `--anchor=true` flag.

My guess is that this line is responsible for this behavior:

https://github.com/terraform-docs/terraform-docs/blob/5f00545aacda52085dfa90adef38c703f2f20579/template/anchor.go#L25

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-name
[^2]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id
[^3]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Identifying_resources_on_the_Web#fragment

**Comment by vale21** [comment]
Can we have an update on this please? Using "name" instead of "id" breakes anchor links on our generated README files. It seems to be a relatively easy fix

**Comment by marrws** [comment]
PR. We now pray to the gods

**Comment by mike-carey** [comment]
If using markdownlint-cli, this breaks [MD051](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md).

```
README.md:8:40 MD051/link-fragments Link fragments should be valid [Context: "[terraform](#requirement\_terraform)"]
README.md:9:34 MD051/link-fragments Link fragments should be valid [Context: "[aws](#requirement\_aws)"]
README.md:15:31 MD051/link-fragments Link fragments should be valid [Context: "[aws](#provider\_aws)"]
...
```

**Comment by marrws** [comment]
> If using markdownlint-cli, this breaks [MD051](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md).
> 
> ```
> README.md:8:40 MD051/link-fragments Link fragments should be valid [Context: "[terraform](#requirement\_terraform)"]
> README.md:9:34 MD051/link-fragments Link fragments should be valid [Context: "[aws](#requirement\_aws)"]
> README.md:15:31 MD051/link-fragments Link fragments should be valid [Context: "[aws](#provider\_aws)"]
> ...
> ```

Are you sure about this?, the rule states:

> This rule is triggered when a link fragment does not match any of the fragments that are automatically generated for headings in a document

Changing `name` to `id` will still link to a fragment not present in the document because the intention here is to have a unique fragment for every variable, not for headers of a document.

IMO, this should be addressed in a separate issue. Maybe adding a `<!-- markdownlint-disable-next-line MD051 -->` on each generated row.

**Comment by mike-carey** [comment]
You are correct @marrws, sorry about that.  Rereading the documentation, it appears that the `name` field should still work.  It looks like it is actually because of the escape character.  When I remove the escape from the links, it no longer gives the error.

Here is the actual issue: https://github.com/DavidAnson/markdownlint/issues/956.  It is within the markdownlint parser itself.
